### PR TITLE
Frontend/enhancement-refactor-code-mobilityPrelude

### PR DESCRIPTION
**Pull request by kavya-shree-s**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Code refactor to have common functions 
Add mobilityPrelude

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+module Engineering.Helpers.MobilityPrelude where
+
+import Data.String (null,Pattern(..), contains, joinWith, toLower, take, toUpper, drop)
+import Data.Maybe (Maybe(..), fromMaybe)
+import PrestoDOM as PD
+import Prelude
+import Data.Array (elem)
+
+isStrEmpty :: String -> Boolean
+isStrEmpty = null
```
Comment (witcher-shailesh) [inline Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs:10]: not required

File: Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+has :: String -> String -> Boolean
+has msg errorPattern = contains (Pattern errorPattern) msg
+
+equalIgnoreCase :: String -> String -> Boolean
+equalIgnoreCase val1 val2 = (toLower val1) == (toLower val2)
```
Comment (witcher-shailesh) [inline Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs:16]: name of function is not intuitive

File: Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+isPresent :: String -> Array String -> Boolean
+isPresent key array =
+      (toLower key) `elem` (map (\item -> toLower item) array)
```
Comment (witcher-shailesh) [inline Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs:20]: what if exact match is required? might not help in our usecase. We can remove

File: Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+-- | Capitalize the first `Char` in a `String`
+capitalize :: String -> String
+capitalize str = (take 1 str # toUpper) <> drop 1 str
+
+-- | Join an `Array` of `String`s with spaces
+unwords :: Array String -> String
+unwords = joinWith " "
```
Comment (witcher-shailesh) [inline Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs:28]: not required

File: Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+toBool :: String -> Boolean
+toBool x = toLower x == "true"
+
+fromMaybeString :: Maybe String -> String
+fromMaybeString = fromMaybe ""
```
Comment (witcher-shailesh) [inline Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs:34]: not required

File: Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+maybeToBool :: Maybe String -> Boolean
+maybeToBool val = toBool $ fromMaybe "" val
+
+maybeArr :: forall a. Maybe (Array a) -> Array a
+maybeArr = fromMaybe []
```
Comment (witcher-shailesh) [inline Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs:40]: not required

File: Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+isJustTrue :: Maybe Boolean -> Boolean
+isJustTrue (Just true) = true
+isJustTrue _ = false
```
Comment (witcher-shailesh) [inline Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs:44]: not required

File: Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+isJustFalse :: Maybe Boolean -> Boolean
+isJustFalse (Just false) = true
+isJustFalse _ = true
```
Comment (witcher-shailesh) [inline Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs:48]: not required

Comment (witcher-shailesh) [reply Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs:48]: just false should be nothing

File: Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs (hunk 1 above)
Comment (kavya-shree-s) [reply Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs:10]: This function is replace to `== ""`

File: Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs (hunk 3 above)
Comment (kavya-shree-s) [reply Frontend/ui-common/src/Engineering/Helpers/MobilityPrelude.purs:20]: Should I rename fiction ? Added few extra functions which we might need later

**Comment by kavya-shree-s** [comment]
Updated in another PR

**Pull request closed without merging**